components/Button/index.js

```diff
@@ -0,0 +1,45 @@
+import { PropTypes } from 'prop-types'
+import { ActivityIndicator, TouchableHighlight } from 'react-native'
+import { Typography } from '../Typography'
+import styles from './styles'
+
+export default function Button ({
+  onPress,
+  title,
+  isLoading,
+  variant = 'primary',
+  containerStyle = {},
+  buttonStyle = {},
+  textStyle = {}
```

Comment:
I'm still trying to wrap my head around how a theme context/provider might help with things like this where we want to override some defaults. I think what we want to do is wrap all the variables and styles for each component into a theme context that can be changed and overridden.

A button is a button normally but sometimes you gotta put one in a weird spot so if we put theming variables and the styles they create into a react context then i think we could deal with the edge cases like...

```
<AppRoot>
  <Provider theme={defaultTheme}>
    <Button />
    <Provider theme={weirdButtonTheme}>
      <Button />
    </Provider>
</AppRoot>
```

I'm pulling a lot of this from [material-ui](https://material-ui.com/customization/theming/) which also includes a good list of [default theme variables](https://material-ui.com/customization/default-theme/)